assets/app/view/game/stock_market.rb

```diff
@@ -242,17 +242,37 @@ def grid_zigzag(zigzag)
       end
 
       def grid_2d
-        @game.stock_market.market.flat_map do |row_prices|
-          row = row_prices.map do |price|
+        # Need to peek at row below to know if sitting on ledge.
+        @game.stock_market.market.push([]).each_cons(2).each_with_index.map do |rows, row_i|
```

Comment:
you don't need to do a flat map?

Reply:
I'm just learning ruby, so please correct me if I'm misunderstanding.

I don't think there's anything to flatten here.  Outer loop is rows, inner loop is columns.  Each column loop creates an array based on the cells, but the row loop doesn't actually see that array, it only gets the `max-content` div.  I don't really know what's going on deep in these `h` objects, but I think `map` and `flat_map` have the same result here.

Reply:
I should amend that to say that `flat_map` and `map` should be expected to produce the same thing, because the result only has 1 axis.

After you posted this, I tried `flat_map` and it doesn't work.  The details of why it doesn't work are beyond me right now, but I'm trying to figure it out.

"  from <embedded>:1523:35:in `constructor.method_missing_stub'"

Reply:
but why did the previous code use a flat_map?

Reply:
I don't think the old code needed it either.  Using the code from master, I printed `grid.size` and `grid.flatten.size` for a handful of 2D markets, and the number was always the same.  I don't have an issue trying to get `flat_map` to work, but I'd need some help with that `method_missing_stub` thing